### PR TITLE
SAAS-1158: mui integration profile

### PR DESCRIPTION
**Pull request by MaksimMalakhovSimbirsoft**
c2abdd905a8.png)

### CODE REVIEW

**Review by nicolalamacchia — APPROVED** [review]

File: src/components/TextInputField/TextInputField.test.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React, { FC } from 'react';
+import { render, screen } from '@testing-library/react';
+import { Form } from 'react-final-form';
+import { TextInputField } from './TextInputField';
+
+export const dataQa = (selector: string) => `[data-qa="${selector}"]`;
+
+export const dataTestId = (selector: string) => `[data-testid="${selector}"]`;
```
Comment (fabio-silva) [inline src/components/TextInputField/TextInputField.test.tsx:8]: Why do we need these two when we have `queryByTestId` and such?

File: src/components/TextInputField/TextInputField.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            value={input.value}
+            id={id}
+            error={!!validationError}
+            helperText={validationError}
+            size={size}
+            placeholder={placeholder}
+            disabled={disabled}
+            fullWidth
+            variant={variant}
+            InputLabelProps={
```
Comment (fabio-silva) [inline src/components/TextInputField/TextInputField.tsx:52]: Probably a better way is to create our own `.d.ts` file where we extend the base typings to allow `data-testid` as part of these props. Check our Grafana fork, `public/app/percona/integrated-alerting/types.ts` or `packages/grafana-ui/src/types/react-table-config.d.ts`

File: src/pages/Profile/Profile.tsx [hunk 3]
```diff
@@ -1,28 +1,48 @@
...
   const dispatch = useDispatch();
   const { pending, email, firstName, lastName } = useSelector(getAuth);
   const [platformAccessToken, setPlatformAccessToken] = useState('');
 
+  const validationSchema = yup.object({
+    firstName: yup.string().label(Messages.firstNameLabel).required().max(50),
+    lastName: yup.string().label(Messages.lastNameLabel).required().max(50),
+  });
+
+  const validation = useCallback(
```
Comment (fabio-silva) [inline src/pages/Profile/Profile.tsx:28]: This seems counter-productive. Do we need to write this piece of code for every single form in our app?

File: src/pages/Profile/Profile.tsx [hunk 4]
```diff
@@ -47,81 +67,89 @@ export const ProfilePage: FC = () => {
 
   return (
     <PrivateLayout>
-      <main className={styles.wrapper}>
+      <Box component="main" sx={styles.wrapper}>
         <Form
           initialValues={{ email, firstName, lastName, platformAccessToken }}
           onSubmit={handleUpdateProfileSubmit}
+          validate={validation()}
```
Comment (fabio-silva) [inline src/pages/Profile/Profile.tsx:74]: This is something we talked about in an earlier PR. The function shouldn't be invoked and your function should have one less arrow. It's `useCallback(async () => {...})`, not `useCallback(() => async() => {})`

File: src/pages/Profile/Profile.tsx [hunk 5]
```diff
@@ -47,81 +67,89 @@ export const ProfilePage: FC = () => {
...
-                <div className={styles.platformAccessTokenButtonWrapper}>
-                  <LinkButton variant="link" onClick={handleCopyToClipboard}>
+                <TextInputField name="lastName" label={Messages.lastNameLabel} margin="normal" size="small" />
+              </Box>
+              <TextInputField
+                fullWidth
+                disabled
+                label={Messages.emailLabel}
+                name="email"
+                margin="normal"
```
Comment (fabio-silva) [inline src/pages/Profile/Profile.tsx:101]: question: isn't "normal" the default?

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: src/components/TextInputField/TextInputField.test.tsx (hunk 1 above)
Comment (MaksimMalakhovSimbirsoft) [reply src/components/TextInputField/TextInputField.test.tsx:8]: @fabio-silva just forgot to remove them

File: src/pages/Profile/Profile.tsx [hunk 6]
```diff
@@ -1,28 +1,48 @@
...
-
 export const ProfilePage: FC = () => {
-  const styles = useStyles(getStyles);
   const dispatch = useDispatch();
   const { pending, email, firstName, lastName } = useSelector(getAuth);
   const [platformAccessToken, setPlatformAccessToken] = useState('');
 
+  const validationSchema = yup.object({
+    firstName: yup.string().label(Messages.firstNameLabel).required().max(50),
+    lastName: yup.string().label(Messages.lastNameLabel).required().max(50),
```
[the remark this answers is not in the document]
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/Profile/Profile.tsx:25]: @fabio-silva Done

File: src/pages/Profile/Profile.tsx (hunk 5 above)
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/Profile/Profile.tsx:101]: @fabio-silva margin by default is 'none'.

File: src/pages/Profile/Profile.tsx (hunk 3 above)
Comment (MaksimMalakhovSimbirsoft) [reply src/pages/Profile/Profile.tsx:28]: @Done, validation function is added to core/utilz

File: src/components/TextInputField/TextInputField.tsx (hunk 2 above)
Comment (MaksimMalakhovSimbirsoft) [reply src/components/TextInputField/TextInputField.tsx]: @fabio-silva we added pages/types/material-ui.ts , added labelProps. But was a problem with helperTextProps and we decided to leave it like it is now as CustomProps

**Comment by fabio-silva** [comment]
@MaksimMalakhovSimbirsoft let's just fix the tests and we're good to do

**Review by fabio-silva — APPROVED** [review]

**Pull request merged**